### ISSUE TITLE
How to run with Outdated Libraries

### ISSUE THREAD

**Issue by Graana128** [Sehaba95/Building_GAN_GUI_Visualizer#2]
pip install torch-scatter -f https://pytorch-geometric.com/whl/torch-1.8.0+cpu.html
pip install torch-sparse -f https://pytorch-geometric.com/whl/torch-1.8.0+cpu.html
pip install torch-cluster -f https://pytorch-geometric.com/whl/torch-1.8.0+cpu.html
pip install torch-spline-conv -f https://pytorch-geometric.com/whl/torch-1.8.0+cpu.html
pip install torch-geometric==1.6.2

**Some of these versions aren't available since torch removed all the versions below 2.0.0**

PS D:\work\graana\Building_GAN_GUI_Visualizer\Building_GAN> pip install torch-scatter -f https://pytorch-geometric.com/whl/torch-1.8.0+cpu.html   
Looking in links: https://pytorch-geometric.com/whl/torch-1.8.0+cpu.html
Collecting torch-scatter
  Using cached torch_scatter-2.1.2.tar.gz (108 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: torch-scatter
  Building wheel for torch-scatter (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\placeholder.py -> build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\scatter.py -> build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\segment_coo.py -> build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\segment_csr.py -> build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\testing.py -> build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\utils.py -> build\lib.win-amd64-cpython-39\torch_scatter
      copying torch_scatter\_init_.py -> build\lib.win-amd64-cpython-39\torch_scatter
      creating build\lib.win-amd64-cpython-39\torch_scatter\composite
      copying torch_scatter\composite\logsumexp.py -> build\lib.win-amd64-cpython-39\torch_scatter\composite
      copying torch_scatter\composite\softmax.py -> build\lib.win-amd64-cpython-39\torch_scatter\composite
      copying torch_scatter\composite\std.py -> build\lib.win-amd64-cpython-39\torch_scatter\composite
      copying torch_scatter\composite\_init_.py -> build\lib.win-amd64-cpython-39\torch_scatter\composite
      running egg_info
      writing torch_scatter.egg-info\PKG-INFO
      writing dependency_links to torch_scatter.egg-info\dependency_links.txt
      writing requirements to torch_scatter.egg-info\requires.txt
      writing top-level names to torch_scatter.egg-info\top_level.txt
      reading manifest file 'torch_scatter.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*' found under directory 'test'
      adding license file 'LICENSE'
      writing manifest file 'torch_scatter.egg-info\SOURCES.txt'
      running build_ext
      c:\users\hp\.pyenv\pyenv-win\versions\3.9.0\lib\site-packages\torch\utils\cpp_extension.py:304: UserWarning: Error checking compiler version for cl: [WinError 2] The system cannot find the file specified
        warnings.warn(f'Error checking compiler version for {compiler}: {error}')
      building 'torch_scatter._scatter_cpu' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
Failed to build torch-scatter
ERROR: Could not build wheels for torch-scatter, which is required to install pyproject.toml-based projects

**Comment by asadrizvi64** [comment]
what python version are you using?

**Comment by asadrizvi64** [comment]
You haven't used the environment variable. Try using python 3.8.0 with pip 21.3.1, try using strong GPU with CUDA else your PC will freeze if you use CPU. Also you might have problems with libraries, try installing torch and geometric first. also one more thing, try using the same PyG as processed data.
